### PR TITLE
scylla-apiclient: drop hk2-locator dependency

### LINKED ISSUE #231: Use newer hk2-locator in order to get rid of javassist (with a known vulnerable version or whatnot)
See https://github.com/eclipse-ee4j/glassfish-hk2/issues/30 - it's fixed in 2.5.0, and indeed, when looking at the deps:
```
[INFO] 
[INFO] --- dependency:2.8:tree (default-cli) @ scylla-apiclient ---
[WARNING] Parameter 'localRepository' is deprecated core expression; Avoid use of ArtifactRepository type. If you need access to local repository, switch to '${repositorySystemSession}' expression and get LRM from it instead.
[INFO] com.scylladb.jmx:scylla-apiclient:jar:1.0
[INFO] +- org.yaml:snakeyaml:jar:2.2:compile
[INFO] +- org.glassfish.jersey.core:jersey-common:jar:2.22.1:compile
[INFO] |  +- (javax.ws.rs:javax.ws.rs-api:jar:2.0.1:compile - omitted for duplicate)
[INFO] |  +- javax.annotation:javax.annotation-api:jar:1.2:compile
[INFO] |  +- org.glassfish.jersey.bundles.repackaged:jersey-guava:jar:2.22.1:compile
[INFO] |  +- org.glassfish.hk2:hk2-api:jar:2.4.0-b31:compile
[INFO] |  |  +- (org.glassfish.hk2:hk2-utils:jar:2.4.0-b31:compile - omitted for conflict with 2.5.0)
[INFO] |  |  \- (org.glassfish.hk2.external:aopalliance-repackaged:jar:2.4.0-b31:compile - omitted for conflict with 2.5.0)
[INFO] |  +- org.glassfish.hk2.external:javax.inject:jar:2.4.0-b31:compile
[INFO] |  +- (org.glassfish.hk2:hk2-locator:jar:2.4.0-b31:compile - omitted for conflict with 2.5.0)
[INFO] |  \- org.glassfish.hk2:osgi-resource-locator:jar:1.0.1:compile
[INFO] +- javax.ws.rs:javax.ws.rs-api:jar:2.0.1:compile
[INFO] +- javax.ws.rs:jsr311-api:jar:1.1.1:compile
[INFO] +- org.glassfish.jersey.core:jersey-client:jar:2.22.1:compile
[INFO] |  +- (javax.ws.rs:javax.ws.rs-api:jar:2.0.1:compile - omitted for duplicate)
[INFO] |  +- (org.glassfish.jersey.core:jersey-common:jar:2.22.1:compile - omitted for duplicate)
[INFO] |  +- (org.glassfish.hk2:hk2-api:jar:2.4.0-b31:compile - omitted for duplicate)
[INFO] |  +- (org.glassfish.hk2.external:javax.inject:jar:2.4.0-b31:compile - omitted for duplicate)
[INFO] |  \- (org.glassfish.hk2:hk2-locator:jar:2.4.0-b31:compile - omitted for duplicate)
[INFO] +- org.glassfish.hk2:hk2-locator:jar:2.5.0:compile
[INFO] |  +- org.glassfish.hk2.external:jakarta.inject:jar:2.5.0:compile
[INFO] |  +- org.glassfish.hk2.external:aopalliance-repackaged:jar:2.5.0:compile
[INFO] |  +- (org.glassfish.hk2:hk2-api:jar:2.5.0:compile - omitted for conflict with 2.4.0-b31)
[INFO] |  +- org.glassfish.hk2:hk2-utils:jar:2.5.0:compile
[INFO] |  |  +- (jakarta.annotation:jakarta.annotation-api:jar:1.3.4:compile - omitted for duplicate)
[INFO] |  |  \- (org.glassfish.hk2.external:jakarta.inject:jar:2.5.0:compile - omitted for duplicate)
[INFO] |  +- jakarta.annotation:jakarta.annotation-api:jar:1.3.4:compile
[INFO] |  \- org.javassist:javassist:jar:3.22.0-CR2:compile
[INFO] +- org.glassfish:javax.json:jar:1.0.4:compile
[INFO] +- com.google.guava:guava:jar:32.1.3-jre:compile
[INFO] |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO] |  +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO] |  +- org.checkerframework:checker-qual:jar:3.37.0:compile
[INFO] |  +- com.google.errorprone:error_prone_annotations:jar:2.21.1:compile
[INFO] |  \- com.google.j2objc:j2objc-annotations:jar:2.8:compile
[INFO] +- com.google.collections:google-collections:jar:1.0:compile
[INFO] +- javax.activation:activation:jar:1.1:compile
[INFO] +- com.fasterxml.jackson.core:jackson-annotations:jar:2.15.3:compile
[INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.15.3:compile
[INFO] |  +- (com.fasterxml.jackson.core:jackson-annotations:jar:2.15.3:compile - omitted for duplicate)
[INFO] |  \- com.fasterxml.jackson.core:jackson-core:jar:2.15.3:compile
[INFO] \- com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:jar:2.15.3:compile
[INFO]    +- com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:jar:2.15.3:compile
[INFO]    |  +- (com.fasterxml.jackson.core:jackson-core:jar:2.15.3:compile - omitted for duplicate)
[INFO]    |  \- (com.fasterxml.jackson.core:jackson-databind:jar:2.15.3:compile - omitted for duplicate)
[INFO]    \- com.fasterxml.jackson.module:jackson-module-jaxb-annotations:jar:2.15.3:compile
[INFO]       +- (com.fasterxml.jackson.core:jackson-annotations:jar:2.15.3:compile - omitted for duplicate)
[INFO]       +- (com.fasterxml.jackson.core:jackson-core:jar:2.15.3:compile - omitted for duplicate)
[INFO]       +- (com.fasterxml.jackson.core:jackson-databind:jar:2.15.3:compile - omitted for duplicate)
[INFO]       +- jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.3:compile
[INFO]       |  \- (jakarta.activation:jakarta.activation-api:jar:1.2.2:compile - omitted for duplicate)
[INFO]       \- jakarta.activation:jakarta.activation-api:jar:1.2.2:compile
```

Vs. original:
```
[ykaul@ykaul scylla-apiclient]$ !mv
mvn dependency:tree -Dverbose=true 
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------< com.scylladb.jmx:scylla-apiclient >------------------
[INFO] Building Scylla REST API client 1.0
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- dependency:2.8:tree (default-cli) @ scylla-apiclient ---
[WARNING] Parameter 'localRepository' is deprecated core expression; Avoid use of ArtifactRepository type. If you need access to local repository, switch to '${repositorySystemSession}' expression and get LRM from it instead.
[INFO] com.scylladb.jmx:scylla-apiclient:jar:1.0
[INFO] +- org.yaml:snakeyaml:jar:2.2:compile
[INFO] +- org.glassfish.jersey.core:jersey-common:jar:2.22.1:compile
[INFO] |  +- (javax.ws.rs:javax.ws.rs-api:jar:2.0.1:compile - omitted for duplicate)
[INFO] |  +- javax.annotation:javax.annotation-api:jar:1.2:compile
[INFO] |  +- org.glassfish.jersey.bundles.repackaged:jersey-guava:jar:2.22.1:compile
[INFO] |  +- org.glassfish.hk2:hk2-api:jar:2.4.0-b31:compile
[INFO] |  |  +- org.glassfish.hk2:hk2-utils:jar:2.4.0-b31:compile
[INFO] |  |  \- org.glassfish.hk2.external:aopalliance-repackaged:jar:2.4.0-b31:compile
[INFO] |  +- org.glassfish.hk2.external:javax.inject:jar:2.4.0-b31:compile
[INFO] |  +- org.glassfish.hk2:hk2-locator:jar:2.4.0-b31:compile
[INFO] |  |  +- (org.glassfish.hk2.external:javax.inject:jar:2.4.0-b31:compile - omitted for duplicate)
[INFO] |  |  +- (org.glassfish.hk2.external:aopalliance-repackaged:jar:2.4.0-b31:compile - omitted for duplicate)
[INFO] |  |  +- (org.glassfish.hk2:hk2-api:jar:2.4.0-b31:compile - omitted for duplicate)
[INFO] |  |  +- (org.glassfish.hk2:hk2-utils:jar:2.4.0-b31:compile - omitted for duplicate)
[INFO] |  |  \- org.javassist:javassist:jar:3.18.1-GA:compile          <-------------------------------------------------------------- XXXXXXXXXXXXX
[INFO] |  \- org.glassfish.hk2:osgi-resource-locator:jar:1.0.1:compile
[INFO] +- javax.ws.rs:javax.ws.rs-api:jar:2.0.1:compile
[INFO] +- javax.ws.rs:jsr311-api:jar:1.1.1:compile
[INFO] +- org.glassfish.jersey.core:jersey-client:jar:2.22.1:compile
[INFO] |  +- (javax.ws.rs:javax.ws.rs-api:jar:2.0.1:compile - omitted for duplicate)
[INFO] |  +- (org.glassfish.jersey.core:jersey-common:jar:2.22.1:compile - omitted for duplicate)
[INFO] |  +- (org.glassfish.hk2:hk2-api:jar:2.4.0-b31:compile - omitted for duplicate)
[INFO] |  +- (org.glassfish.hk2.external:javax.inject:jar:2.4.0-b31:compile - omitted for duplicate)
[INFO] |  \- (org.glassfish.hk2:hk2-locator:jar:2.4.0-b31:compile - omitted for duplicate)
[INFO] +- org.glassfish:javax.json:jar:1.0.4:compile
[INFO] +- com.google.guava:guava:jar:32.1.3-jre:compile
[INFO] |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO] |  +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO] |  +- org.checkerframework:checker-qual:jar:3.37.0:compile
[INFO] |  +- com.google.errorprone:error_prone_annotations:jar:2.21.1:compile
[INFO] |  \- com.google.j2objc:j2objc-annotations:jar:2.8:compile
[INFO] +- com.google.collections:google-collections:jar:1.0:compile
[INFO] +- javax.activation:activation:jar:1.1:compile
[INFO] +- com.fasterxml.jackson.core:jackson-annotations:jar:2.15.3:compile
[INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.15.3:compile
[INFO] |  +- (com.fasterxml.jackson.core:jackson-annotations:jar:2.15.3:compile - omitted for duplicate)
[INFO] |  \- com.fasterxml.jackson.core:jackson-core:jar:2.15.3:compile
[INFO] \- com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:jar:2.15.3:compile
[INFO]    +- com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:jar:2.15.3:compile
[INFO]    |  +- (com.fasterxml.jackson.core:jackson-core:jar:2.15.3:compile - omitted for duplicate)
[INFO]    |  \- (com.fasterxml.jackson.core:jackson-databind:jar:2.15.3:compile - omitted for duplicate)
[INFO]    \- com.fasterxml.jackson.module:jackson-module-jaxb-annotations:jar:2.15.3:compile
[INFO]       +- (com.fasterxml.jackson.core:jackson-annotations:jar:2.15.3:compile - omitted for duplicate)
[INFO]       +- (com.fasterxml.jackson.core:jackson-core:jar:2.15.3:compile - omitted for duplicate)
[INFO]       +- (com.fasterxml.jackson.core:jackson-databind:jar:2.15.3:compile - omitted for duplicate)
[INFO]       +- jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.3:compile
[INFO]       |  \- (jakarta.activation:jakarta.activation-api:jar:1.2.2:compile - omitted for duplicate)
[INFO]       \- jakarta.activation:jakarta.activation-api:jar:1.2.2:compile
[INFO] ------------------------------------------------------------------------

```

Untested patch:
```
[ykaul@ykaul scylla-apiclient]$ git diff
diff --git a/scylla-apiclient/pom.xml b/scylla-apiclient/pom.xml
index 7667afe..7a92cf6 100644
--- a/scylla-apiclient/pom.xml
+++ b/scylla-apiclient/pom.xml
@@ -42,6 +42,11 @@
             <artifactId>jersey-client</artifactId>
             <version>2.22.1</version>
         </dependency>
+       <dependency>
+            <groupId>org.glassfish.hk2</groupId>
+            <artifactId>hk2-locator</artifactId>
+            <version>2.5.0</version>
+        </dependency>
         <dependency>
             <groupId>org.glassfish</groupId>
             <artifactId>javax.json</artifactId>
```

### PR DESCRIPTION
**Pull request by tchaikov**
Drop the dependency of hk2-locator, in order to get rid of javaassist and other 3rd party dependencies of it.

there are two ways to address this problem:

1. bump up the dependencies which depend on hk2-locator to a version which depend on hk2-locator 2.5.0. because hk2-locator 2.5.0 contains a change to drop the unnecessary dependencies which made their way into the BOM. but they should have not.
2. bump up the dependencies which depend on hk2-locator to a version which does not depend on hk2-locator at all.

before this change, per the output of `mvn dependency:tree -Dverbose=true`, we indirectly depend on hk2-locator 2.4.0.

after this change, hk2-locator dependency is dropped by bumping up org.glassfish.jersey.core to the oldest stable version which was released (see https://mvnrepository.com/artifact/org.glassfish.jersey.core/jersey-common/2.27) after hk2-locator 2.5.0 was released (see
https://mvnrepository.com/artifact/org.glassfish.hk2/hk2-locator/2.5.0-b42), otherwise we still depend on hk2-locator 2.4.0 indirectly.

verified by running

```shell
mvn dependency:tree -Dverbose=true | grep hk2-locator
```
nothing shows up with this change.

Fixes #231
Signed-off-by: Kefu Chai <kefu.chai@scylladb.com>

### CODE REVIEW

**Review by yaronkaikov — APPROVED** [review]
LGTM

**Comment by tchaikov** [comment]
being tested at https://github.com/scylladb/scylladb/pull/16680

**Comment by mykaul** [comment]
+- (javax.ws.rs:javax.ws.rs-api:jar:2.1:compile - omitted for conflict with 2.0.1)

**Comment by tchaikov** [comment]
> +- (javax.ws.rs:javax.ws.rs-api:jar:2.1:compile - omitted for conflict with 2.0.1)

v2:

- bump up javax.ws.rs-api as well.

**Comment by tchaikov** [comment]
@yaronkaikov @mykaul thank you for your reviews. could you take another look?

**Comment by mykaul** [comment]
The only way I can review this is see the output of ```mvn dependency:tree -Dverbose=true``` on this patch and see that we are now better than before.
then we have to somehow run CI on this to ensure we did not break anything, but that's an extra step later on, I reckon.

**Comment by mykaul** [comment]
Also, see https://github.com/scylladb/java-driver/blob/085dd34ef46b1de5111629bdccfe4056c4355676/pom.xml#L74 - unsure, but it makes sense that we'll align...

**Comment by tchaikov** [comment]
> The only way I can review this is see the output of `mvn dependency:tree -Dverbose=true` on this patch and see that we are now better than before. then we have to somehow run CI on this to ensure we did not break anything, but that's an extra step later on, I reckon.

it's running at https://github.com/scylladb/scylladb/pull/16680

**Comment by tchaikov** [comment]
> Also, see https://github.com/scylladb/java-driver/blob/085dd34ef46b1de5111629bdccfe4056c4355676/pom.xml#L74 - unsure, but it makes sense that we'll align...

unless it's urgent, i'd leave it to some one who actually understands Java better than me.

**Comment by mykaul** [comment]
@tchaikov - indeed - https://stackoverflow.com/questions/44088493/jersey-stopped-working-with-injectionmanagerfactory-not-found - that's what I was worried about all along.

**Comment by tchaikov** [comment]
v3:

- add `jersey-hk2`

**Comment by tchaikov** [comment]
@mykaul hi Yaniv, the test at https://github.com/scylladb/scylladb/pull/16680 is green now. could you take another look?

**Comment by tchaikov** [comment]
```console
$ mvn dependency:tree -Dverbose=true        
[INFO] Scanning for projects...                                                                                        
[WARNING]                                                                                                              
[WARNING] Some problems were encountered while building the effective model for com.scylladb.jmx:scylla-jmx:jar:1.0
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 61, column 21
[WARNING]                                                                                                              
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING]                                                                                                              
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING]                                                                                                              
[INFO]                                                                                                                 
[INFO] --------------------< com.scylladb.jmx:scylla-jmx >---------------------                            
[INFO] Building Scylla JMX 1.0                                                                                         
[INFO]   from pom.xml                                                                                                  
[INFO] --------------------------------[ jar ]---------------------------------                        
[INFO]                                                                                                                 
[INFO] --- dependency:2.8:tree (default-cli) @ scylla-jmx ---             
[WARNING] Parameter 'localRepository' is deprecated core expression; Avoid use of ArtifactRepository type. If you need access to local repository, switch to '${repositorySystemSession}' expression and get LRM from it instead.
[INFO] com.scylladb.jmx:scylla-jmx:jar:1.0                                                                             
[INFO] +- com.scylladb.jmx:scylla-apiclient:jar:1.0:compile
[INFO] |  +- org.yaml:snakeyaml:jar:2.2:compile     
[INFO] |  +- org.glassfish.jersey.core:jersey-common:jar:2.27:compile          
[INFO] |  |  +- (javax.ws.rs:javax.ws.rs-api:jar:2.1:compile - omitted for duplicate)
[INFO] |  |  +- javax.annotation:javax.annotation-api:jar:1.2:compile          
[INFO] |  |  +- org.glassfish.hk2.external:javax.inject:jar:2.5.0-b42:compile
[INFO] |  |  \- org.glassfish.hk2:osgi-resource-locator:jar:1.0.1:compile
[INFO] |  +- org.glassfish.jersey.core:jersey-client:jar:2.27:compile          
[INFO] |  |  +- (javax.ws.rs:javax.ws.rs-api:jar:2.1:compile - omitted for duplicate)
[INFO] |  |  +- (org.glassfish.jersey.core:jersey-common:jar:2.27:compile - omitted for duplicate)
[INFO] |  |  \- (org.glassfish.hk2.external:javax.inject:jar:2.5.0-b42:compile - omitted for duplicate)
[INFO] |  +- org.glassfish.jersey.inject:jersey-hk2:jar:2.27:compile
[INFO] |  |  +- (org.glassfish.jersey.core:jersey-common:jar:2.27:compile - omitted for duplicate)
[INFO] |  |  \- org.glassfish.hk2:hk2-locator:jar:2.5.0-b42:compile
[INFO] |  |     +- (org.glassfish.hk2.external:javax.inject:jar:2.5.0-b42:compile - omitted for duplicate)
[INFO] |  |     +- org.glassfish.hk2.external:aopalliance-repackaged:jar:2.5.0-b42:compile
[INFO] |  |     +- org.glassfish.hk2:hk2-api:jar:2.5.0-b42:compile
[INFO] |  |     |  +- javax.inject:javax.inject:jar:1:compile
[INFO] |  |     |  +- (org.glassfish.hk2:hk2-utils:jar:2.5.0-b42:compile - omitted for duplicate)
[INFO] |  |     |  \- (org.glassfish.hk2.external:aopalliance-repackaged:jar:2.5.0-b42:compile - omitted for duplicate) 
[INFO] |  |     +- org.glassfish.hk2:hk2-utils:jar:2.5.0-b42:compile
[INFO] |  |     |  +- (javax.annotation:javax.annotation-api:jar:1.2:compile - omitted for duplicate)
[INFO] |  |     |  \- (javax.inject:javax.inject:jar:1:compile - omitted for duplicate)
[INFO] |  |     +- (javax.annotation:javax.annotation-api:jar:1.2:compile - omitted for duplicate)
[INFO] |  |     \- org.javassist:javassist:jar:3.22.0-CR2:compile
[INFO] |  +- javax.ws.rs:javax.ws.rs-api:jar:2.1:compile
[INFO] |  +- javax.ws.rs:jsr311-api:jar:1.1.1:compile
[INFO] |  +- org.glassfish:javax.json:jar:1.0.4:compile
[INFO] |  +- com.google.guava:guava:jar:32.1.3-jre:compile
[INFO] |  |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |  |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO] |  |  +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO] |  |  +- org.checkerframework:checker-qual:jar:3.37.0:compile
[INFO] |  |  +- com.google.errorprone:error_prone_annotations:jar:2.21.1:compile
[INFO] |  |  \- com.google.j2objc:j2objc-annotations:jar:2.8:compile
[INFO] |  +- com.google.collections:google-collections:jar:1.0:compile
[INFO] |  +- javax.activation:activation:jar:1.1:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-annotations:jar:2.15.3:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.15.3:compile
[INFO] |  |  +- (com.fasterxml.jackson.core:jackson-annotations:jar:2.15.3:compile - omitted for duplicate)
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-core:jar:2.15.3:compile
[INFO] |  \- com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:jar:2.15.3:compile
[INFO] |     +- com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:jar:2.15.3:compile
[INFO] |     |  +- (com.fasterxml.jackson.core:jackson-core:jar:2.15.3:compile - omitted for duplicate)
[INFO] |     |  \- (com.fasterxml.jackson.core:jackson-databind:jar:2.15.3:compile - omitted for duplicate)
[INFO] |     \- com.fasterxml.jackson.module:jackson-module-jaxb-annotations:jar:2.15.3:compile
[INFO] |        +- (com.fasterxml.jackson.core:jackson-annotations:jar:2.15.3:compile - omitted for duplicate)
[INFO] |        +- (com.fasterxml.jackson.core:jackson-core:jar:2.15.3:compile - omitted for duplicate)
[INFO] |        +- (com.fasterxml.jackson.core:jackson-databind:jar:2.15.3:compile - omitted for duplicate)
[INFO] |        +- jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.3:compile
[INFO] |        |  \- (jakarta.activation:jakarta.activation-api:jar:1.2.2:compile - omitted for duplicate)
[INFO] |        \- jakarta.activation:jakarta.activation-api:jar:1.2.2:compile
[INFO] \- junit:junit:jar:4.13.1:test
[INFO]    \- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.762 s
[INFO] Finished at: 2024-01-09T10:54:37+08:00
[INFO] ------------------------------------------------------------------------
```

**Comment by mykaul** [comment]
That looks clean now, thanks.

**Comment by tchaikov** [comment]
@scylladb/scylla-jmx-maint hello maintainers, could you help merge this change?

**Comment by denesb** [comment]
Submodule update: scylladb/scylladb@5981900dca4da0de3f1072061296041f83b410cf

**Pull request closed without merging**
